File: quorum-test-network/dapps/quorumToken/test-contract.js
// Script para verificar o contrato implantado
const { ethers } = require("hardhat");

async function checkContract() {
  // Endereço do contrato implantado
  const contractAddress = "0x5FbDB2315678afecb367f032d93F642f64180aa3";
  
  // Obter o ABI do contrato
  const contractABI = require("./artifacts/contracts/TicketNFT.sol/TicketNFT.json").abi;
  
  // Obter contrato
  const contract = await ethers.getContractAt("TicketNFT", contractAddress);
  
  try {
    // Tentar obter o proprietário
    const owner = await contract.owner();
    console.log("Proprietário do contrato:", owner);
    
    // Tentar obter o total supply
    const totalSupply = await contract.totalSupply();
    console.log("Total Supply:", totalSupply.toNumber());
    
    console.log("Contrato está respondendo corretamente!");
  } catch (error) {
    console.error("Erro ao interagir com o contrato:", error);
  }
}

checkContract().catch(console.error);

File: quorum-test-network/dapps/quorumToken/backend/test_contract.py
import os
from web3 import Web3
from dotenv import load_dotenv
import json

# Carregar .env da pasta raiz
load_dotenv('../.env')

# Configurações
RPC_URL = os.getenv("RPC_URL", "http://localhost:8545")
CONTRACT_ADDRESS = os.getenv("CONTRACT_ADDRESS")

# Caminho para o ABI
abi_path = './TicketNFT.json'

# Conectar ao blockchain
w3 = Web3(Web3.HTTPProvider(RPC_URL))

# Carregar ABI
try:
    with open(abi_path, 'r') as f:
        contract_data = json.load(f)
        abi = contract_data['abi'] if 'abi' in contract_data else contract_data
except FileNotFoundError:
    print(f"Arquivo ABI não encontrado: {abi_path}")
    exit(1)

# Criar instância do contrato
try:
    contract = w3.eth.contract(address=CONTRACT_ADDRESS, abi=abi)
    print(f"Instância do contrato criada com sucesso para: {CONTRACT_ADDRESS}")
    
    # Testar uma chamada de leitura (view function)
    try:
        owner = contract.functions.owner().call()
        print(f"Proprietário do contrato: {owner}")
    except Exception as e:
        print(f"Erro ao chamar função 'owner': {e}")
    
    # Testar totalSupply
    try:
        total_supply = contract.functions.totalSupply().call()
        print(f"Total Supply: {total_supply}")
    except Exception as e:
        print(f"Erro ao chamar função 'totalSupply': {e}")

    # Testar se o contrato responde
    try:
        block_number = w3.eth.block_number
        print(f"Bloco mais recente: {block_number}")
    except Exception as e:
        print(f"Erro ao obter bloco: {e}")

except Exception as e:
    print(f"Erro ao criar instância do contrato: {e}")